src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,127 @@
-import { useState } from 'react';
+import { FC, useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
+import { pattern } from '../../utils/validationPatterns';
 
-export const NewMovie = () => {
+interface Props {
+  onAdd: (movie: Movie) => void;
+}
+
+export const NewMovie: FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [titleValue, setTitleValue] = useState('');
+  const [descriptionValue, setDescriptionValue] = useState('');
+  const [imgUrlValue, setImgUrlValue] = useState('');
+  const [imdbUrlValue, setImdbUrlValue] = useState('');
+  const [imdbIdValue, setImdbIdValueValue] = useState('');
+
+  // eslint-disable-next-line max-len
+  const isFieldFilled = titleValue.trim() && imgUrlValue.trim() && imdbIdValue.trim() && imdbUrlValue.trim();
+
+  const isUrlValid = (urlValue: string) => {
+    return pattern.test(urlValue);
+  };
+
+  const handleSubmit = (event: { preventDefault: () => void; }) => {
```

Comment:
just get type to the event
```suggestion
  const handleSubmit = (event: ...) => {
```